src/torch_models.py

```diff
@@ -508,6 +517,25 @@ def _predict_sample_once(self, x: Array) -> Array:
         sample_idx = torch.argmax(scores)
         return sample_ys[sample_idx]
 
+    def _predict_grid(self, x: Array) -> Array:
+        assert self._grid_num_ticks_per_dim is not None
+        assert self._grid_num_ticks_per_dim > 0
+        dy = 1.0 / self._grid_num_ticks_per_dim
+        ticks = [np.arange(0.0, 1.0, dy)] * self._y_dim
+        grid = np.meshgrid(*ticks)  # type: ignore
+        candidate_ys = np.transpose(grid).reshape((-1, self._y_dim))
```

Comment:
isn't the grid symmetric? what does this transpose do?

Reply:
without the transpose, the shape would be (num_samples, y_dim) instead of (y_dim, num_samples)